src/raftstore/store/snap.rs

```diff
@@ -1023,18 +1056,54 @@ impl SnapManager {
                 format!("{} should be a directory", path.display()),
             ));
         }
+
         for f in fs::read_dir(path)? {
             let p = f?;
+            // In the older version all snapshot file are in the same folder.
+            // This code migrate old snapshot files to current file layout
+            // that put files of each snapshot to its own folder
             if p.file_type()?.is_file() {
                 if let Some(s) = p.file_name().to_str() {
                     if s.ends_with(TMP_FILE_SUFFIX) {
+                        // .tmp files are deleted at starting stage
+                        // in old version, so just delete it here
                         fs::remove_file(p.path())?;
-                    } else if s.ends_with(SST_FILE_SUFFIX) {
+                        continue;
+                    }
+                    let (dirname, filename) = Self::gen_migrate_target(s);
+                    let dirpath = path.join(dirname);
+                    let filepath = dirpath.join(filename);
+                    create_dir_if_not_exist(&dirpath)?;
+                    sync_dir(&dirpath)?;
+                    fs::rename(p.path(), &filepath)?;
+                    sync_dir(&filepath)?;
```

Comment:
Can we use another function name, since here actually we are syncing a file.

Reply:
This is a mistake, the file itself do not need sync, should sync path and dirpath instead.